docs/Using Randomness/vrf-best-practices.md

```diff
@@ -0,0 +1,57 @@
+---
+layout: nodes.liquid
+section: smartContract
+date: Last Modified
+title: "VRF Best Practices"
+permalink: "docs/chainlink-vrf-best-practices/"
+hidden: false
+metadata: 
+  title: "Chainlink VRF API Reference"
+  description: "Best pracices for using Chainlink VRF."
+  image: 
+    0: "/files/OpenGraph_V3.png"
+    1: "670379d-OpenGraph_V3.png"
+    2: 1459
+    3: 1459
+    4: "#dbe1f8"
+---
+
+Best are the practices for using Chainlink VRF.
+
+## 1. Getting a random number within a range
+
+If you need to generate a random number within a given range, you should use [modulo](https://docs.soliditylang.org/en/v0.5.3/types.html#modulo) to define the limits of your range. Below you can see how to get a random number between 1 and 50.
+
+```solidity
+uint256 public randomResult;
+
+function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
+    randomResult = (randomness % 50) + 1;
+}
+```
+
+## 2. Getting multiple random numbers
+
+If you want to get multiple random numbers from a single VRF response, you should create an array where the `randomValue` is your original returned VRF number, and `n` is the desired number of random numbers.
+
+```solidity
+function expand(uint256 randomValue, uint256 n) public pure returns (uint256[] memory expandedValues) {
+    expandedValues = new uint256[](n);
+    for (uint256 i = 0; i < n; i++) {
+        expandedValues[i] = uint256(keccak256(abi.encode(randomValue, i)));
+    }
+    return expandedValues;
+}
+```
+
+## 3. Having multiple VRF requests in flight
+
+If you want to have multiple VRF requests in flight, you should create a mapping between the request ID and the actual random number.
+
+```solidity
+mapping(bytes32 => uint256) requestIdToRandomNumber;
+
```

Comment:
You need to add this mapping in the initial request as well. Please change this to:
If you want to have multiple VRF requests in flight, you might want to create a mapping between the request ID and the requester.

Or, maybe you want to keep the ordering of when a request was made, in that case, you might want to use a mapping of `requestId` to index. 
```
mapping(bytes32 => address) public requestIdToAddress;
mapping(bytes32 => uint256) public requestIdToRequestNumber;
uint256 public counter;

function getRandomNumber(uint256 userProvidedSeed) public returns (bytes32 requestId) {
        bytes32 requestId =  requestRandomness(keyHash, fee, userProvidedSeed);
        requestIdToAddress[requestId] = msg.sender;
        requestIdToRequestNumber[requestId] = counter;
        counter += 1;
    }

function fulfillRandomness(bytes32 requestId, uint256 randomness) internal override {
    address from = requestIdToAddress[requestId];
    uint256 requestNumber = requestIdToRandomNumberIndex[requestId];
}
```
Feel free to use whatever data structure you prefer.

Reply:
Added. Did we have any requests for keeping a track of multiple VRF results based on request ID?

Reply:
No, we could add that in too if you think it makes sense.